internal/error.go

```diff
@@ -350,6 +350,16 @@ func (e *ContinueAsNewError) Error() string {
 	return "ContinueAsNew"
 }
 
+// WorkflowFunc return workflow function of the new run
+func (e *ContinueAsNewError) WorkflowFunc() interface{} {
+	return e.wfn
```

Comment:
One example:
For migrating continue-as-new workflows, workflow interceptor can instead complete and start a new workflow with the same workflow options in the new domain. In the startworkflow request, it needs the workflow function rather than workflow type.

Reply:
After discussion, this is actually not needed. WorklowType alone is enough